File: backend/src/main/java/com/yash/taskmanager/service/TaskService.java
package com.yash.taskmanager.service;

import com.yash.taskmanager.dto.TaskRequest;
import com.yash.taskmanager.entity.Task;

import java.util.List;

public interface TaskService {

    Task createTask(TaskRequest request);

    List<Task> getTasks();

    Task updateTask(Long id, TaskRequest request);

    void deleteTask(Long id);
}

File: frontend/src/services/taskService.js
import api from "../api/axiosConfig";

export const getTasks =
  () => api.get("/tasks");

export const createTask =
  (task) =>
    api.post(
      "/tasks",
      task
    );

export const updateTask =
  (id, task) =>
    api.put(
      `/tasks/${id}`,
      task
    );

export const deleteTask =
  (id) =>
    api.delete(
      `/tasks/${id}`
    );